app/src/common/chromium/com/igalia/wolvic/browser/api/impl/TabMediaSessionObserver.java

```diff
@@ -0,0 +1,281 @@
+package com.igalia.wolvic.browser.api.impl;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.Paint;
+import android.os.SystemClock;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.igalia.wolvic.browser.api.WImage;
+import com.igalia.wolvic.browser.api.WMediaSession;
+import com.igalia.wolvic.browser.api.WResult;
+
+import org.chromium.base.task.PostTask;
+import org.chromium.content_public.browser.MediaSession;
+import org.chromium.content_public.browser.MediaSessionObserver;
+import org.chromium.content_public.browser.UiThreadTaskTraits;
+import org.chromium.content_public.browser.WebContents;
+import org.chromium.components.browser_ui.media.MediaNotificationImageUtils;
+import org.chromium.components.browser_ui.media.MediaImageCallback;
+import org.chromium.components.browser_ui.media.MediaImageManager;
+import org.chromium.services.media_session.MediaImage;
+import org.chromium.services.media_session.MediaMetadata;
+import org.chromium.services.media_session.MediaPosition;
+
+import java.util.List;
+
+import org.chromium.media_session.mojom.MediaSessionAction;
+
+public class TabMediaSessionObserver extends MediaSessionObserver implements MediaImageCallback {
+    private static final int UPDATE_POSITION_TIME_MS = 500;
```

Comment:
Where does this 500ms come from?

Reply:
This comes from my prediction. :)
Chrome browser uses a 250 ms period and manages a progress bar based on the milliseconds unit.
Wolvic manages it based on the seconds unit, so I thought it is enough to update the position with a 500 ms period. But I don't have a strong opinion here.